### PR TITLE
DataLabelWidget: More date options

### PR DESCRIPTION
**Pull request by apepper**
Allows to format a date value as "Date and time" and "Only date". "Relative" was previously already available.

Needs content: https://edit.scrivito.com/d0a154d76edf2a7bd991fc658e700a1d~https://datalabelwidget-date.scrivito-portal-app.pages.dev/my-tynacoon?_scrivito_workspace_id=mc2dbbd39b44e23a&_scrivito_display_mode=view

### CODE REVIEW

File: src/Widgets/DataLabelWidget/DataLabelWidgetEditingConfig.ts [hunk 1]
```diff
@@ -17,11 +18,27 @@ provideEditingConfig(DataLabelWidget, {
...
   },
-  properties: ['attributeName', 'showAs', 'valueSize', 'marginBottom'],
+  properties: (widget) =>
+    [
+      'attributeName',
+      'showAs',
+      widget.get('showAs') === 'datetime' ? 'datetimeFormat' : null,
+      'valueSize',
+      'marginBottom',
+    ].filter(isString),
```
Comment (dcsaszar) [inline src/Widgets/DataLabelWidget/DataLabelWidgetEditingConfig.ts:41]: * [ ] FYI I'm skeptical about adding dependencies for trivial stuff like `].filter(p => p !== null),`

File: src/utils/formatDate.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+const dateTimeFormatLocale = 'en-GB'
+
+export function formatDateTime(date: Date) {
```
Comment (dcsaszar) [inline src/utils/formatDate.ts:3]: * [ ] FYI Refactoring idea: Export a type for available formats and provide a single helper function.

```ts
export type DateFormat = 'dateTime' | 'fullDateTime' // | etc
export function formatDate(date: Date, format: DateFormat)
```

**Review by dcsaszar — APPROVED** [review]
LGTM

File: src/Widgets/DataLabelWidget/DataLabelWidgetEditingConfig.ts (hunk 1 above)
Comment (apepper) [reply src/Widgets/DataLabelWidget/DataLabelWidgetEditingConfig.ts:41]: I already tried several things, e.g.

```ts
  properties: (widget): string[] =>
    [
      'attributeName',
      'showAs',
      widget.get('showAs') === 'datetime' ? 'datetimeFormat' : null,
      'valueSize',
      'marginBottom',
    ].filter((p) => p !== null),
```

or
```ts
  properties: (widget): string[] =>
    [
      'attributeName',
      'showAs',
      widget.get('showAs') === 'datetime' ? 'datetimeFormat' : null,
      'valueSize',
      'marginBottom',
    ].filter((p) => typeof p === 'string'),
```

All of these still print:

```
Type '(string | null)[]' is not assignable to type 'string[]'.
Type 'string | null' is not assignable to type 'string'.
Type 'null' is not assignable to type 'string'.
```

With `isString` (or probably some other guard function) it works.

Comment (dcsaszar) [reply src/Widgets/DataLabelWidget/DataLabelWidgetEditingConfig.ts:41]: I see, TS stuff.
The cleanest I can think of right now is an inline type guard:
```ts
].filter((p): p is string => typeof p === 'string'),
```

Comment (apepper) [reply src/Widgets/DataLabelWidget/DataLabelWidgetEditingConfig.ts:41]: => https://github.com/Scrivito/scrivito-portal-app/pull/333

**Pull request merged**